src/svgdevicecontext.cpp

```diff
@@ -814,7 +814,7 @@ void SvgDeviceContext::DrawLine(int x1, int y1, int x2, int y2)
 {
     pugi::xml_node pathChild = AddChild("path");
     pathChild.append_attribute("d") = StringFormat("M%d %d L%d %d", x1, y1, x2, y2).c_str();
-    if (m_penStack.top().HasColor()) {
+    if (m_penStack.top().HasColor() || this->UseGlobalStyling()) {
```

Comment:
This does not make sense to me. It seems it would now use `m_penStack.top.GetColor()` even when `m_penStack.top.HasColor()` is `false`

Reply:
Your design:

> case (COLOR_NONE): return "currentColor";

Reply:
Your comment seems to imply that *my* design is bad. Would you be able to explain why, and how you would change it? That would be helpful.

Reply:
I'm not feeling very chatty today, sorry. Your design is not bad. 
When `--mm-output` is used `UseGlobalStyling` is set to false and we need to add `@stroke`. 

So looking again I see the problem.

```suggestion
    if (m_penStack.top().HasColor() || !this->UseGlobalStyling()) {
```

Do you have a better solution?